pkg/build/build.go

```diff
@@ -350,6 +361,51 @@ func pullImage(ctx *buildContext, image string) (id string, err error) {
 	return pushTarget.Context().Digest(digest.String()).String(), nil
 }
 
+func resolveLocalImage(ctx *buildContext, image string) (string, error) {
+	c, err := k8sclient.Default()
+	if err != nil {
+		return "", err
+	}
+
+	// very important - make sure we only list images in the buildNamespace to avoid finding ones in other projects
+	imageList := v1.ImageInstanceList{}
+	err = c.List(ctx.ctx, &imageList, &kclient.ListOptions{
+		Namespace: ctx.buildNamespace,
+	})
+	if err != nil {
+		return "", err
+	}
+
+	var digest string
+imageLoop:
+	for _, imageInstance := range imageList.Items {
```

Comment:
Could this be replaced by the `FindImageMatch` function that's also used by `client.ImageGet`?
https://github.com/acorn-io/acorn/blob/main/pkg/images/find.go#L39

Reply:
I'll go try it

Reply:
I guess there's some easy way to translate from ImageInstance to Image, since they're basically the same struct-wise, but I'm not sure if it's worth the effort now.

Reply:
Well it turns out my assumption was wrong that I wouldn't be able to get the public Acorn API types. So I just updated it to use Images instead of ImageInstances and to use the FindImageMatch function. Thanks for the suggestion - glad we don't need to duplicate the logic of that function